src/components/sets/sets.js

```diff
@@ -586,13 +603,17 @@ export default class SetsTree {
   emitVisibilityUpdate() {
     if (this.onVisibilityChange) {
       let cellIds = [];
+      const cellColors = {};
       this.visibleKeys.forEach((setKey) => {
         const node = this.findNode(setKey);
         if (node && node.set && node.set.length > 0) {
           cellIds = [...cellIds, ...node.set];
+          node.set.forEach((cellId) => {
+            cellColors[cellId] = node.color;
+          });
```

Comment:
Slight preference for pulling this out and iterating over the completed `cellIds` instead... Nesting the loops can make things look more complicated than they really are.